### PR TITLE
[DOC] Adding doc folder and tightening makefile recipes for checking style

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Addresses #20 and setting up docs folder.

We need to migrate to MNE to setup circleci for docs building, so it'll take a few iterations to make sure we got everything right w/o errors.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-connectivity/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by adam2392** [comment]
FYI: @mscheltienne

File: doc/install.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+------------
+
+* ``mne`` (>=1.0)
+* ``numpy`` (>=1.14)
+* ``scipy`` (>=1.5.0)
+* ``joblib`` (>=1.0.0)
+* ``scikit-learn`` (>=1.1)
+* ``torch`` (for running pytorch neural networks)
+* ``python-picard`` (for running ICA)
+* ``matplotlib`` (optional, for using the interactive data inspector)
```
Comment (mscheltienne) [inline doc/install.rst:16]: Minor detail for now, but this should be updated when #16 is resolved. I'm guessing matplotlib and joblib are not required?

File: doc/references.bib [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+@book{OppenheimEtAl1999,
+  address = {Upper Saddle River, NJ},
+  edition = {2 edition},
+  title = {Discrete-{Time} {Signal} {Processing}},
+  isbn = {978-0-13-754920-7},
+  publisher = {Prentice Hall},
+  author = {Oppenheim, Alan V. and Schafer, Ronald W. and Buck, John R.},
+  year = {1999}
+}
```
Comment (mscheltienne) [inline doc/references.bib:22]: To be removed?

**Comment by adam2392** [comment]
Gonna merge to make the dev process easier w/ the Makefile recipes.

File: doc/api.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+:py:mod:`mne_icalabel`:
+
+.. automodule:: mne_icalabel
+   :no-members:
+   :no-inherited-members:
+
+This is the application programming interface (API) reference
+for classes (``CamelCase`` names) and functions
+(``underscore_case`` names) of MNE-Connectivity, grouped thematically by analysis
```
Comment (mscheltienne) [inline doc/api.rst:13]: One MNE-Connectivity reference that slipped by ;)

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+[tool.black]
+line-length = 100
```
Comment (mscheltienne) [inline pyproject.toml:2]: I think it says `115` in `.flake8`. Normal?

File: doc/references.bib (hunk 2 above)
Comment (adam2392) [reply doc/references.bib:22]: Oh yeah good catch, I need to replace this with the ICLabel publication and add to the docstring in #21 for `label_ic` and the `ICLabelNet`.

File: pyproject.toml (hunk 4 above)
Comment (adam2392) [reply pyproject.toml:2]: Not sure tbh. It might be needed to not error out, but I'll do some tests.

**Pull request merged**